Core/include/Acts/Definitions/Algebra.hpp

```diff
@@ -0,0 +1,102 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+// for GNU: ignore this specific warning, otherwise just include Eigen/Dense
+#if defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmisleading-indentation"
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#pragma GCC diagnostic pop
+#else
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#endif
+
+namespace Acts {
+
+/// This is the Scalar used for all EDM and Geometry based object,
+/// can be customised.
+#ifdef ACTS_CUSTOM_SCALARTYPE
+using ActsScalar = ACTS_CUSTOM_SCALARTYPE;
+#else
+using ActsScalar = double;
+#endif
```

Comment:
Ah I see the default is here. It might be 'better' if we just put the default in a CMake option, and just do 

```cpp
using ActsScalar = ACTS_CUSTOM_SCALARTYPE;
```

unconditionally. Then the definition and default are all in one place, and it should effectively do the same thing.

Reply:
I see, I will change that accordingly.

Reply:
To summarize my argument for leaving the current implementation as-is: this would introduce (for the first time) a non-optional preprocessor definition without which the code can not be used. This adds a (small) complication without giving us any functionality or feature. The behavior of the code would be identical regardless of the way it is implemented, so I see no necessity to change it.

While CMake can set the required default value for us, I can image that someone installs Acts as an external dependency via CMake, but then use a different build system for their own software. By switching to a required preprocessor definition, this add additional road-blocks. I agree that these are only small obstacles, but it feels to me more fragile than the current implementation.

I wanted to mention my arguments here, but I will leave it up to @asalzburger to decide.